code/game/machinery/autolathe.dm

```diff
@@ -369,15 +369,16 @@
 	var/mass_per_sheet = 0 // Amount of material constituting one sheet.
 
 	for(var/obj/O in eating.GetAllContents(includeSelf = TRUE))
-		if(O.matter && O.matter.len)
-			for(var/material in O.matter)
+		var/list/_matter = O.get_matter()
+		if(_matter && _matter)
```

Comment:
This is a duplicate check. Do not check for `.len` before calling `for` either, it checks automatically.